bib/cmd/bootc-image-builder/main.go

```diff
@@ -100,7 +102,59 @@ func loadConfig(path string) (*BuildConfig, error) {
 	return &conf, nil
 }
 
+func getRootFSType(imgref string) (string, error) {
+	var buffer bytes.Buffer
+	configCmd := exec.Command("podman", "run", "--rm", imgref, "bootc", "install", "print-configuration")
+	configCmd.Stdout = &buffer
+
+	if err := configCmd.Run(); err != nil {
+		return "", fmt.Errorf("failed to run bootc install print-configuration: %w", err)
+	}
+
+	var bootcConfig struct {
+		Filesystem struct {
+			Root struct {
+				Type string `json:"type"`
+			} `json:"root"`
+		} `json:"filesystem"`
+	}
+
+	if err := json.Unmarshal(buffer.Bytes(), &bootcConfig); err != nil {
+		return "", fmt.Errorf("failed to unmarshal bootc configuration: %w", err)
+	}
+
+	// filesystem.root.type is the preferred way instead of the old root-fs-type top-level key.
+	// See https://github.com/containers/bootc/commit/558cd4b1d242467e0ffec77fb02b35166469dcc7
+	fsType := bootcConfig.Filesystem.Root.Type
+	supportedFS := []string{"ext4", "xfs"}
```

Comment:
I'm curious about the allowlist here; it seems distribution dependent at the minimum right?  E.g. some distros/OSes use btrfs, there's also bcachefs coming in, etc.

Maybe it'd be better to have a denylist instead, i.e. not allow doing `vfat` for example.  We could I guess do a sanity check that `mkfs.$type` exists in the root I guess to give a better error perhaps?

Reply:
I wonder if this is the right place for policy like this (i.e. I would rather just remove it). It seems to me the best place for this would be something like a "bootc lint" or "bootc build" that does checks. If we add the policy here we need to duplicate it accross `bootc install to-disk` and `anaconda` so having it in one central place seems more maintainable to me?